addons/website/models/website_menu.py

```diff
@@ -136,6 +137,26 @@ def clean_url(self):
                     url = '/%s' % self.url
         return url
 
+    def is_active(self):
+        if not request:
+            return False
+
+        # Don't compare to `url` as it could be shadowed by the linked website
+        # page's URL
+        menu_url = self.clean_url()
+        if not menu_url:
+            return False
+
+        menu_url = unslug_url(menu_url)
+        request_url = unslug_url(request.httprequest.path)
+        if request_url == menu_url:
+            return True
+
+        if menu_url != '/' and any(request_url == unslug_url(child.url) for child in self.child_id if child.url):
+            return True
```

Comment:
Maybe a comment above this one. Not entirely sure what it handles: dropdown-toggles ?

Reply:
There is one in next commit as you mentioned at https://github.com/odoo/odoo/pull/107782/commits/25ea3e7f20edc435fd933d084b3adadb13c03412#r1059395918 (on top of a full docstring)
But I'll move it in the first commit